packages/datadog-instrumentations/src/mocha/worker.js

```diff
@@ -0,0 +1,199 @@
+const { addHook, channel, AsyncResource } = require('../helpers/instrument')
```

Comment:
ℹ️ file for the mocha worker: it's a smaller version of `mocha/main.js`, with support for tests only

Reply:
Since as you mentioned it's basically the same as `main.js`, any way to reuse most of the code? There seems to be quite a lot of duplication between the two otherwise.

Reply:
there are small differences, but I guess I could extract _some_ mocha event handlers like `this.on('test', testHandler`). I'll give that a try, thanks 👍

Reply:
done